R/variable_label.R

```diff
@@ -0,0 +1,346 @@
+      , name = rep("label", length(ordered_value))
+      , value = ordered_value
+      , USE.NAMES = TRUE
+      , SIMPLIFY = FALSE
+    )
+    d <- as.data.frame(
+      d
+      , col.names = names(modified_object)
+      , stringsAsFactors = FALSE
+    )
+    object[, columns_to_annotate] <- d
+
+    object
+  }
+)
+
+
+
+#' @keywords internal
+
+setGeneric(
+  name = "assign_annotation"
+  , def = function(object, value, name){
+    standardGeneric("assign_annotation")
+  }
+)
+
+
+
+#' @keywords internal
+
+setMethod(
+  "assign_annotation"
+  , signature = c(object = "annotated_vector", value = "character", name = "character")
+  , definition = function(object, value, name){
+    slot(object@annotation, name = name) <- value
+    object@label <- object@annotation@label
+    object
+  }
+)
+
+#' #' @keywords internal
+#'
+#' setMethod(
+#'   "assign_annotation"
+#'   , signature = c(object = "vector", value = "character", name = "character")
+#'   , definition = function(object, value, name){
+#'     object <- new(
+#'       paste0("annotated_", class(object))
+#'       , annotation <- new("vector_annotation")
+#'     )
+#'     object@label <- object@annotation@label
+#'     object# return
+#'   }
+#' )
+
+
+
+#' @title Set default variable labels from column names
+#' @description We use this function internally to provide default variable for all columns in a data.frame from column names.
+#' @param x A \code{data.frame}
+#' @return Returns a \code{data.frame} with labelled columns. Labels are preserved (if already specified), otherwise generated from column names.
+#' @rdname default_label
+#' @keywords internal
+
+setGeneric(
+  "default_label"
+  , def = function(object){
+    standardGeneric("default_label")
+  }
+)
+
+
+
+#' @rdname default_label
+#' @keywords internal
+
+setMethod(
+  "default_label"
+  , signature = "data.frame"
+  , definition = function(object){
+    columns <- sapply(X = variable_label(object), FUN = is.null, simplify = TRUE)
+
+    if(any(columns)){
+      value <- colnames(object)[columns]
+      names(value) <- value
+      variable_label(object[, columns]) <- value
+    }
+    object
+  }
+)
+
+
+
+#' Combine to expression
+#'
+#' We use this internal function to generate expressions that can be used for plotting. Accepts a list of elements that are coerced,
+#' currently supperted elements are \code{character}, \code{expression}, and \code{character} that contain \code{latex} elements.
+#'
+#' @param x A \code{list} that contains all elements that are intended to be coerced into one expression.
+#' @return An expression
+#' @keywords internal
+
+combine_plotmath <- function(x){
+
+  x <- lapply(X  = x, FUN = tex_conv)
+  y <- as.expression(substitute(paste(a, b), list(a = x[[1]], b = x[[2]])))
+
+  if(length(x)>2){
+    for (i in 3:length(x)){
+      y <- as.expression(substitute(paste(a, b), list(a = y[[1]], b = x[[i]])))
+    }
+  }
+  return(y)
+}
+
+
+
+#' @importFrom latex2exp TeX
```

Comment:
Do we really need to import this function here? Shouldn't it be possible to keep it in `Suggests` similar to `dplyr` for `fast_aggregate()`?

Reply:
Yes, I think everything will nicely work even if it is not installed (the function already has a parameter checking if `latex2exp` is installed). Just removed the import and moved to suggests.